arch/arm/core/CMakeLists.txt

```diff
@@ -21,3 +21,4 @@ zephyr_sources_ifdef(CONFIG_USERSPACE userspace.S)
 
 add_subdirectory_ifdef(CONFIG_CPU_CORTEX_M cortex_m)
 add_subdirectory_ifdef(CONFIG_CPU_HAS_MPU  cortex_m/mpu)
+add_subdirectory_ifdef(CONFIG_CPU_CORTEX_M_HAS_CMSE cortex_m/cmse)
```

Comment:
Does this really need to be a subdir, do we expect more files in here?

Reply:
@galak, since this is specific to ARMv8-M, I wouldn't prefer to "pollute" the cortex-m/ directory, which is generic to all Cortex-M  MCUs; that's why I created this new sub-directory.

It might get a Kconfig, later, too, as we will bring-in the secure part of the API, so it gets really cleaner IMO with a separate folder.

Reply:
I also think it make sense, especially given that there's already an `mpu` folder